src/main/java/com/radixdlt/crypto/Hash.java

```diff
@@ -216,9 +220,7 @@ public boolean equals(Object o) {
 
 		if (o instanceof Hash) {
 			Hash other = (Hash) o;
-
-			// `hashCode()` uses `this.hashCodeCached`, which in turn is derived from `this.data`.
-			return this.hashCode() == other.hashCode();
+			return Arrays.equals(this.data, other.data);

```

Comment:
I think that this should go back to previous checks, eg
```
return this.hashCode() == other.hashCode() && Arrays.equals(this.data, other.data);
```

This is a major win in performance given that hashCode value is precomputed, and only requires the complete array scan/compare in cases where there are hashCode collisions.

I think it would also be super prudent to include a comment, more or less the words in the preceding paragraph so that we aren't tempted to reintroduce the issue at a later date.